desktop/ui/src/main/java/org/datacleaner/documentation/template/DocumentationCreator.java

```diff
@@ -0,0 +1,211 @@
+import freemarker.template.Configuration;
+import freemarker.template.MalformedTemplateNameException;
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+import freemarker.template.TemplateNotFoundException;
+
+public class DocumentationCreator {
+
+    private static final String HTMLBASE64_PREFIX = "data:image/png;base64,";
+    private static final File cssFile = new File("documentationReference.css");
+    private static final String FILENAME_TEMPLATE = "documentationTemplate.html";
+
+    @SuppressWarnings("deprecation")
+    private static final Configuration freemarkerConfiguration = new Configuration();
+    private final Map<String, Object> _data = new HashMap<String, Object>();;
+    private static final Logger logger = LoggerFactory.getLogger(DocumentationCreator.class);
+    private Template _template;
+
+    public DocumentationCreator() {
+        freemarkerConfiguration.setClassForTemplateLoading(this.getClass(), "/");
+    }
+
+    /**
+     * Creates the documentation reference.
+     * 
+     * @param componentdescriptor
+     * @param outputStream
+     * @throws IOException
+     * @throws ParseException
+     * @throws MalformedTemplateNameException
+     * @throws TemplateNotFoundException
+     */
+    public void createDocumentation(ComponentDescriptor<?> componentdescriptor, OutputStream outputStream)
+            throws TemplateNotFoundException, MalformedTemplateNameException, ParseException, IOException {
+
+        final Template template = getTemplate();
+
+        try {
+            {
+                _data.put("cssPath",cssFile.getPath());
+            }
+            {
+                _data.put("component", componentdescriptor);
+                final Set<ProvidedPropertyDescriptor> providedProperties = componentdescriptor.getProvidedProperties();
+                _data.put("providedproperties", providedProperties);
+
+            }
+            {
+                final Set<ConfiguredPropertyDescriptor> configuredProperties = componentdescriptor
+                        .getConfiguredProperties();
+
+                if (configuredProperties != null) {
+                    final List<ConfiguredPropertyDescriptor> properties = new ArrayList<ConfiguredPropertyDescriptor>(
+                            configuredProperties);
+                    _data.put("properties", properties);
+                    Map<String, MappedProperty> mappedProperties = new HashMap<String, MappedProperty>();
+                    for (ConfiguredPropertyDescriptor property : properties) {
+                        final MappedProperty mappedProperty = property.getAnnotation(MappedProperty.class);
+                        if (mappedProperty != null) {
+                            mappedProperties.put(property.getName(), mappedProperty);
+                        }
+                    }
+                    _data.put("mappedproperties", mappedProperties);
+                }
```

Comment:
More stuff we should have info on in the documentation:

 * The result time (if the component produces results)
 * The output column types (if the component is a transformer)
 * The filter outcomes (if the component is a filter)